cli/deploy.ts

```diff
@@ -132,15 +135,22 @@ const createSchemas = async (schemaNames: string[]) => {
           } else if (status.isInBlock || status.isFinalized) {
             const evt = eventWithSectionAndMethod(events, "schemas", "SchemaCreated");
             if (evt) {
-              const val = evt?.data[1];
-              console.log("SUCCESS: " + schemaName + " schema created with id of " + val);
-              resolve({ schemaName, id: Number(val.toHuman()) });
-            } else resolve({ schemaName });
+              const id = evt?.data[1];
```

Comment:
maybe not a big deal for your purposes but do you need to unsubscribe too?

Reply:
I think so, for correctness. Probably doesn't matter too much since it's a run-and-done CLI program, but I wouldn't be surprised if there are some error logs that come out when the program is done that would be cleaned up with an `unsub()` call. I'll pull this branch and see...

Reply:
I see no errors when I run; like I said, probably not a critical issue since this is CLI run-and-done. Fix if you like; non-blocking.

Reply:
I don't think we're actually subscribed here -- if the event is not immediately in the block or finalized, it looks like we never resolve or reject the Promise. Obviously this seems to work fine in an ideal environment (especially one with instant sealing) but I don't think it handles edge cases or busy chains very well.

I'd prefer to make that a separate issue though as it's a pre-existing condition, and maybe someone slightly more familiar with how to subscribe and poll could add this or give me some pointers.

Reply:
Added as #49